app/packages/core/src/components/Modal/Group.tsx

```diff
@@ -175,12 +173,8 @@ const MainSample: React.FC<{
     )
       return undefined;
 
-    if (currentModalSlice === defaultSlice) {
-      return allSlices.find((s) => s !== defaultSlice);
-    }
-
-    return defaultSlice;
-  }, [currentModalSlice, defaultSlice, sample, allSlices]);
+    return allSlices.filter((slice) => slice !== currentModalSlice)[0];
```

Comment:
This will be cleaned up in sample routing. But the modal group slice is guaranteed to be defined when this modal is rendered as it is set before opening the modal in `useSetExpandedSample`